src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java

```diff
@@ -368,7 +368,7 @@ public boolean hasMaxColumnValueLengths()
         @Override
         public boolean hasImplicitlyFrozenTuples()
         {
-            return false;
+            return version.compareTo("me") <= 0;
```

Comment:
(also the wording "explicit" in TrieIndexFormat when setting its implementation of this method is off)

Reply:
I know the work "explicit" was off; I was going to fix that but forgot 🤦🏻

Reply:
wording fixed in https://github.com/datastax/cassandra/pull/1440/commits/32bb558b526d85d220785d29c2592d8a245f7b5a

Reply:
I must admit that I didn't pay attention to BIG format when working on that patch.

Let's analyze a bit - I'm not sure but I conclude it worked like this in the past:
| Version   | Format | non-frozen UDT | dropped non-frozen UDT | frozen UDT | dropped frozen UDT |
| --------- | ------ | ---------- | ----------- | ----------- |----------- |
| 3.0 - 3.5 | `ma`   | not supported | not supported | udt | udt |
| 3.6 | `ma` | udt | udt | frozen udt | frozen udt |
| 3.7 - 3.11 | `mb` - `me` | udt | udt | frozen udt | frozen udt |
| 4.0+ | `na`... | udt | not supported | frozen udt | tuple |

Tuples were always considered implicitly frozen in OSS. However, it is not true in DSE.

This flag, if `false`, makes the implicit frozen invalid and such an sstable is reject on load.

In particular, the validation may fail in case of:

```java
if (isMultiCell()) 
{
    if (isTuple() && !isDroppedColumn && !isForOfflineTool)
        throw columnException(columnName,
                              "tuple type %s is not frozen, which should not have happened",
                              asCQL3Type().toSchemaString());
```
that is, multicell tuple, which is not a dropped column - such a thing does not exist in any Cassandra version. Since, such a column may not exist, Cassandra skips writing explicit `frozen` modifier for tuple types in serialization headers.

and the fix is applied if `hasImplicitlyFrozenTuples` and:
```java
                if (invalidType.isMultiCell())
                {
                    boolean isMultiCell = !invalidType.isTuple() || isDroppedColumn;
                    return invalidType.with(AbstractType.freeze(invalidType.subTypes()), isMultiCell);
```

which means, if the type is multi-cell (implied), it's multi-cell flag is set to true if `isTuple => isDropped`. That is, it will be changed from multi-cell to frozen iff it is a tuple and it is not a dropped column (because the only place were we expect non-frozen tuples is the list of dropped columns). 

Concluding, there is no BIG format version coming from OSS which has explicit freezing of tuples. DSE could produce N sstables with that but BIG format is not used in DSE. 

In other words, this methods should just return `true` for BGI format.

Reply:
The comment has been corrected.   And `big` formats always return `true`.


One problem we have in HCD is for `big` sstables is we don't know if formats were written with C* or DSE.
e.g. the `me` format can be from either C* 3.11 or from DSE 5.1
as far as this fix, and possible regressions, goes it is not important (as we're flooring sstable compatibility to `me`)